### DISCUSSION TITLE
missing dynamics representation?

### DISCUSSION THREAD

**Discussion by sbowman-mitre** [TUDelft-CNS-ATM/openap#13] [General]
Hi, thanks for you work on this open-source api. I do have some confusion that I'm hoping you can clarify. In your paper announcing OpenAP (July 2020, [mdpi](https://www.mdpi.com/2226-4310/7/8/104)) I see 4dof equations of motion discussed for aircraft dynamics. However, I do not see this represented in the code of this repo. In [tray.py](https://github.com/junzis/openap/blob/master/openap/traj/gen.py) I see climb, descent, and complete trajectory generation capabilities, but they do not represent dynamics, only kinematics. How is dynamics represented in the code? If it isn't yet here, when do you expect to include that capability? Thanks! 

Of course, send me a PM is you want to discuss further offline.

**Comment by junzis** [comment]
Hi @sbowman-mitre, you are correct. The generation of trajectory is only based on the kinematics model. The idea here is to use the kinematic model for generating semi-realistic trajectories (e.g. for fast time simulation). Currently, the dynamic model is used implicitly for calculating drag, fuel flow, and emissions. 

One improvement we can add to trajectory generation is to allow uses to specify the aircraft mass. Then we would need to use the dynamic model to ensure the only flyable trajectories are generated.

I see you also raised another question regarding optimization in a different thread. That will be another place where the dynamic model will be used extensively. 

I hope this somehow answers your question. And let's keep the conversation here online, so others can also see it in the future :)

**Reply by sbowman-mitre** [reply]
Okay, thanks for clarifying. But, I'm still wondering about the plan or roadmap that you have in mind. Do you expect to add dynamics at some point? Or will that be in a different module (e.g. BlueSky)? Or are you looking for others to implement these and provide PRs?

**Reply by sbowman-mitre** [reply]
Circling back on this. I wonder if you'd be interested in an open-source collaboration with some of us at MITRE? We may have something you could use and we find your work of value to us also. We have an open-source dynamics and control model (FIM MOPS Aircraft and Control Model) that is used by manufacturers for avionics certification purposes. The model is documented [here](https://www.mitre.org/publications/technical-papers/derivation-of-a-point-mass-aircraft-model-used-for-fast-time) and the c++ implementation is [on GitHub here](https://mitre.github.io/FMACM/). The implementation abstracts the FMS guidance logic away to an input file (allowing guidance solutions to be specific to the user). It currently uses BADA for aircraft performance and dynamics, but we're interested in connecting OpenAP also. Let me know what you think.

**Reply by junzis** [reply]
Yes, certainly, you are very welcomed to connecting OpenAP model/data. Please also keep an eye on the updates. Feel free to get in contact with me directly by email if you have any questions regarding that.

I haven't spent much effort on the roadmap. My idea is to go with the flow, and see where the ATM community takes with it :)
